include/ginkgo/core/distributed/matrix.hpp

```diff
@@ -399,7 +408,7 @@ class Matrix
      *              The default is the MPI_COMM_WORLD.
      */
     explicit Matrix(std::shared_ptr<const Executor> exec,
-                    mpi::communicator comm = mpi::communicator(MPI_COMM_WORLD));
+                    mpi::communicator comm);
```

Comment:
these are redundant and may lead to people accidentally passing the wrong executor in the communicator?

Reply:
I guess you mean that we should use `MPI_Comm` instead? Because otherwise a user could not use a specific communicator.

Reply:
or just take an mpi::communicator. Though I'm not 100% happy with the ergonomics of that either